### ISSUE TITLE
[Bug]: Netdata always core dumped, cannot start

### ISSUE THREAD

**Issue by Woart** [netdata/netdata#14909]
### Bug description

current version:
`netdata v1.38.0-380-nightly`

```
# dmesg -T|grep netdata|more
[Fri Apr 14 07:19:52 2023] netdata[1867736]: segfault at 0 ip 000056534e230e52 sp 00007ffeaa426e30 error 4 in netdata[56534e07b000+3ff000]
[Fri Apr 14 07:20:26 2023] netdata[1868110]: segfault at 0 ip 00005566d1882e52 sp 00007fff0f968170 error 4 in netdata[5566d16cd000+3ff000]
[Fri Apr 14 07:20:58 2023] netdata[1868304]: segfault at 0 ip 00005642cae79e52 sp 00007fff981a9400 error 4 in netdata[5642cacc4000+3ff000]
[Fri Apr 14 07:21:29 2023] netdata[1868510]: segfault at 0 ip 0000562276977e52 sp 00007ffca76fddf0 error 4 in netdata[5622767c2000+3ff000]
[Fri Apr 14 07:22:01 2023] netdata[1868703]: segfault at 0 ip 00005613cdb05e52 sp 00007ffee6b57770 error 4 in netdata[5613cd950000+3ff000]
[Fri Apr 14 07:22:32 2023] netdata[1868927]: segfault at 0 ip 000055e33a584e52 sp 00007fff43211030 error 4 in netdata[55e33a3cf000+3ff000]
[Fri Apr 14 07:23:04 2023] netdata[1869137]: segfault at 0 ip 00005644794ffe52 sp 00007fffcb739960 error 4 in netdata[56447934a000+3ff000]
[Fri Apr 14 07:23:36 2023] netdata[1869334]: segfault at 0 ip 0000559dc6671e52 sp 00007fff842945e0 error 4 in netdata[559dc64bc000+3ff000]
[Fri Apr 14 07:24:07 2023] netdata[1869531]: segfault at 0 ip 000055c3a4bbfe52 sp 00007ffda0dd8de0 error 4 in netdata[55c3a4a0a000+3ff000]
[Fri Apr 14 07:24:39 2023] netdata[1869728]: segfault at 0 ip 000055ca452f7e52 sp 00007ffcac603a00 error 4 in netdata[55ca45142000+3ff000]
[Fri Apr 14 07:25:10 2023] netdata[1869968]: segfault at 0 ip 000055e250814e52 sp 00007ffe72f313d0 error 4 in netdata[55e25065f000+3ff000]
[Fri Apr 14 07:25:42 2023] netdata[1870162]: segfault at 0 ip 000055f822ffde52 sp 00007ffea10cdb30 error 4 in netdata[55f822e48000+3ff000]
[Fri Apr 14 07:26:14 2023] netdata[1870359]: segfault at 0 ip 000055d2e445de52 sp 00007fffcfbbe2c0 error 4 in netdata[55d2e42a8000+3ff000]
[Fri Apr 14 07:26:45 2023] netdata[1870559]: segfault at 0 ip 0000559313d55e52 sp 00007ffd52c58a30 error 4 in netdata[559313ba0000+3ff000]
[Fri Apr 14 07:27:17 2023] netdata[1870766]: segfault at 0 ip 000055ca7939ce52 sp 00007ffcbdfc2d80 error 4 in netdata[55ca791e7000+3ff000]
[Fri Apr 14 07:27:48 2023] netdata[1870963]: segfault at 0 ip 000055de4ca2be52 sp 00007ffcc7dac580 error 4 in netdata[55de4c876000+3ff000]
[Fri Apr 14 07:28:20 2023] netdata[1871190]: segfault at 0 ip 000055b885c68e52 sp 00007ffe5850c8f0 error 4 in netdata[55b885ab3000+3ff000]
[Fri Apr 14 07:28:52 2023] netdata[1871390]: segfault at 0 ip 000055d4a9193e52 sp 00007fff66660650 error 4 in netdata[55d4a8fde000+3ff000]
[Fri Apr 14 07:29:23 2023] netdata[1871590]: segfault at 0 ip 000055898aa43e52 sp 00007ffce8392af0 error 4 in netdata[55898a88e000+3ff000]
[Fri Apr 14 07:29:54 2023] netdata[1871787]: segfault at 0 ip 00005577f3105e52 sp 00007ffc8f060250 error 4 in netdata[5577f2f50000+3ff000]
[Fri Apr 14 07:30:26 2023] netdata[1871980]: segfault at 0 ip 00005625b0d45e52 sp 00007ffcda7b66e0 error 4 in netdata[5625b0b90000+3ff000]
[Fri Apr 14 07:30:58 2023] netdata[1872180]: segfault at 0 ip 00005629eea32e52 sp 00007fff63be39a0 error 4 in netdata[5629ee87d000+3ff000]
[Fri Apr 14 07:31:29 2023] netdata[1872415]: segfault at 0 ip 000055858401de52 sp 00007ffe2604f640 error 4 in netdata[558583e68000+3ff000]
[Fri Apr 14 07:32:01 2023] netdata[1872609]: segfault at 0 ip 0000565131f0be52 sp 00007fffa023cbe0 error 4 in netdata[565131d56000+3ff000]
[Fri Apr 14 07:32:33 2023] netdata[1872811]: segfault at 0 ip 00005575d95c6e52 sp 00007ffcb5352350 error 4 in netdata[5575d9411000+3ff000]
[Fri Apr 14 07:33:04 2023] netdata[1873012]: segfault at 0 ip 00005640475e9e52 sp 00007ffe992c5190 error 4 in netdata[564047434000+3ff000]
[Fri Apr 14 07:33:36 2023] netdata[1873212]: segfault at 0 ip 0000559a5e626e52 sp 00007fff59329090 error 4 in netdata[559a5e471000+3ff000

# rm -rf /var/cache/netdata
# service netdata start

# ps ax|grep netdata
1952883 pts/1    S+     0:00 grep --color=auto netdata

# tail -100 /var/log/netdata/error.log
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_HOST_OS_VERSION_ID=20.04
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_HOST_OS_DETECTION=/etc/os-release
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_HOST_IS_K8S_NODE=false
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_KERNEL_NAME=Linux
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_KERNEL_VERSION=5.4.0-29-generic
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_ARCHITECTURE=x86_64
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_VIRTUALIZATION=kvm
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_VIRT_DETECTION=systemd-detect-virt
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CONTAINER=none
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CONTAINER_DETECTION=systemd-detect-virt
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CPU_LOGICAL_CPU_COUNT=4
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CPU_VENDOR=GenuineIntel
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CPU_MODEL=Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CPU_FREQ=2599000000
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_CPU_DETECTION=lscpu
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_TOTAL_RAM=4126883840
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_RAM_DETECTION=procfs
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_TOTAL_DISK_SIZE=85899345920
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_SYSTEM_DISK_DETECTION=sysfs
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_INSTANCE_CLOUD_TYPE=unknown
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_INSTANCE_CLOUD_INSTANCE_TYPE=unknown
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA_INSTANCE_CLOUD_INSTANCE_REGION=unknown
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA STARTUP: in    1191 ms, collecting system info - next: initialize RRD structures
2023-04-14 10:52:45: netdata INFO  : MAIN : SQLite database /var/cache/netdata/netdata-meta.db initialization
2023-04-14 10:52:45: netdata INFO  : MAIN : metadata database version is 7 (no migration needed)
2023-04-14 10:52:45: netdata INFO  : MAIN : SQLite database initialization completed
2023-04-14 10:52:45: netdata INFO  : MAIN : SQLite database /var/cache/netdata/context-meta.db initialization
2023-04-14 10:52:45: netdata INFO  : MAIN : context database version is 1 (no migration needed)
2023-04-14 10:52:45: netdata INFO  : MAIN : CONFIG: cannot load user config '/etc/netdata/stream.conf'. Will try stock config.
2023-04-14 10:52:45: netdata INFO  : MAIN : DBENGINE: Initializing ...
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread created with task id 1953028
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread created with task id 1953029
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread created with task id 1953030
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : set name of thread 1953030 to DBENGINE_INIT
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : set name of thread 1953028 to DBENGINE_INIT
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : set name of thread 1953029 to DBENGINE_INIT
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: found 2 files in path /var/cache/netdata/dbengine-tier2
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: found 2 files in path /var/cache/netdata/dbengine
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: found 2 files in path /var/cache/netdata/dbengine-tier1
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: initializing data file "/var/cache/netdata/dbengine-tier2/datafile-1-0000000001.ndf".
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: initializing data file "/var/cache/netdata/dbengine-tier1/datafile-1-0000000001.ndf".
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: initializing data file "/var/cache/netdata/dbengine/datafile-1-0000000001.ndf".
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: data file "/var/cache/netdata/dbengine-tier1/datafile-1-0000000001.ndf" initialized (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: data file "/var/cache/netdata/dbengine-tier2/datafile-1-0000000001.ndf" initialized (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: data file "/var/cache/netdata/dbengine/datafile-1-0000000001.ndf" initialized (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: loading journal file '/var/cache/netdata/dbengine-tier1/journalfile-1-0000000001.njf'
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: journal file '/var/cache/netdata/dbengine-tier1/journalfile-1-0000000001.njf' loaded (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: loading journal file '/var/cache/netdata/dbengine-tier2/journalfile-1-0000000001.njf'
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: journal file '/var/cache/netdata/dbengine-tier2/journalfile-1-0000000001.njf' loaded (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: populating retention to MRG from 1 journal files of tier 1, using 1 threads...
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: populating retention to MRG from 1 journal files of tier 2, using 1 threads...
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread with task id 1953029 finished
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread with task id 1953030 finished
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: loading journal file '/var/cache/netdata/dbengine/journalfile-1-0000000001.njf'
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: journal file '/var/cache/netdata/dbengine/journalfile-1-0000000001.njf' loaded (size:4096).
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : DBENGINE: populating retention to MRG from 1 journal files of tier 0, using 1 threads...
2023-04-14 10:52:45: netdata INFO  : DBENGINE_INIT : thread with task id 1953028 finished
2023-04-14 10:52:45: netdata INFO  : MAIN : DBENGINE: tier 0 is ready for data collection and queries
2023-04-14 10:52:45: netdata INFO  : MAIN : DBENGINE: tier 1 is ready for data collection and queries
2023-04-14 10:52:45: netdata INFO  : MAIN : DBENGINE: tier 2 is ready for data collection and queries
2023-04-14 10:52:45: netdata INFO  : MAIN : Starting metadata sync thread with 1024 entries command queue
2023-04-14 10:52:45: netdata INFO  : MAIN : SQLite metadata sync initialization complete
2023-04-14 10:52:45: netdata INFO  : MAIN : Host '<HOST>' (at registry as '<HOST>') with guid 'bfb2d500-afff-11ed-b0ee-0ffe29aef61d' initialized, os 'linux', timezone 'Etc/UTC', tags '', program_name 'netdata', program_version 'v1.38.0-380-nightly', update every 1, memory mode dbengine, history entries 0, streaming disabled (to '' with api key ''), health disabled, cache_dir '/var/cache/netdata', alarms default handler '', alarms default recipient ''
2023-04-14 10:52:45: netdata INFO  : MAIN : Creating archived hosts
2023-04-14 10:52:45: netdata INFO  : MAIN : Created 0 archived hosts
2023-04-14 10:52:45: netdata INFO  : MAIN : ACLK sync initialization completed
2023-04-14 10:52:45: netdata INFO  : MAIN : Starting ACLK synchronization thread
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA STARTUP: in      12 ms, initialize RRD structures - next: check for incomplete shutdown
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA STARTUP: in       0 ms, check for incomplete shutdown - next: collect claiming info
2023-04-14 10:52:45: netdata INFO  : MAIN : Unable to load '/var/lib/netdata/cloud.d/claimed_id', setting state to AGENT_UNCLAIMED
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA STARTUP: in       0 ms, collect claiming info - next: collect host labels
2023-04-14 10:52:45: netdata INFO  : MAIN : NETDATA STARTUP: in       6 ms, collect host labels - next: start the static threads
2023-04-14 10:52:45: netdata INFO  : P[idlejitter] : thread created with task id 1953061
2023-04-14 10:52:45: netdata INFO  : P[idlejitter] : set name of thread 1953061 to P[idlejitter]
2023-04-14 10:52:45: netdata INFO  : STATS_SQLITE3 : thread created with task id 1953065
2023-04-14 10:52:45: netdata INFO  : STATS_SQLITE3 : set name of thread 1953065 to STATS_SQLITE3
2023-04-14 10:52:45: netdata INFO  : STATS_GLOBAL : thread created with task id 1953063
2023-04-14 10:52:45: netdata INFO  : STATS_GLOBAL : set name of thread 1953063 to STATS_GLOBAL
2023-04-14 10:52:45: netdata INFO  : STATS_WORKERS : thread created with task id 1953064
2023-04-14 10:52:45: netdata INFO  : STATS_WORKERS : set name of thread 1953064 to STATS_WORKERS
2023-04-14 10:52:45: netdata INFO  : HEALTH : thread created with task id 1953062
2023-04-14 10:52:45: netdata INFO  : HEALTH : set name of thread 1953062 to HEALTH
2023-04-14 10:52:45: netdata INFO  : PLUGINSD : thread created with task id 1953066
2023-04-14 10:52:45: netdata INFO  : PLUGINSD : set name of thread 1953066 to PLUGINSD
2023-04-14 10:52:45: netdata INFO  : STATSD_FLUSH : thread created with task id 1953068
2023-04-14 10:52:45: netdata INFO  : STATSD_FLUSH : set name of thread 1953068 to STATSD_FLUSH
2023-04-14 10:52:45: netdata INFO  : EXPORTING : thread created with task id 1953069
2023-04-14 10:52:45: netdata INFO  : EXPORTING : set name of thread 1953069 to EXPORTING
2023-04-14 10:52:45: netdata INFO  : ACLK_MAIN : thread created with task id 1953070
2023-04-14 10:52:45: netdata INFO  : ACLK_MAIN : set name of thread 1953070 to ACLK_MAIN
2023-04-14 10:52:45: netdata INFO  : REPLAY[1] : thread created with task id 1953072
2023-04-14 10:52:45: netdata INFO  : REPLAY[1] : set name of thread 1953072 to REPLAY[1]
2023-04-14 10:52:45: netdata INFO  : ACLK_MAIN : Waiting for Cloud to be enabled
2023-04-14 10:52:45: netdata INFO  : RRDCONTEXT : thread created with task id 1953071
2023-04-14 10:52:45: netdata INFO  : RRDCONTEXT : set name of thread 1953071 to RRDCONTEXT
2023-04-14 10:52:45: netdata INFO  : P[proc] : thread created with task id 1953073
2023-04-14 10:52:45: netdata INFO  : P[proc] : set name of thread 1953073 to P[proc]
2023-04-14 10:52:45: netdata INFO  : EXPORTING : Processing connector instance (<server>)
2023-04-14 10:52:45: netdata INFO  : EXPORTING : Instance (<server>) on connector (opentsdb:http) is enabled and scheduled for activation
2023-04-14 10:52:45: netdata INFO  : EXPORTING : Instance <server> on opentsdb:http

# dmesg -T|tail -

[4694716.330093] netdata[1953274]: segfault at 0 ip 0000558c2936ce52 sp 00007ffed0c43950 error 4 in netdata[558c291b7000+3ff000]
[4694716.330101] Code: 0f 1f 84 00 00 00 00 00 48 8b 1d 29 e7 3b 00 4c 89 f2 4c 89 ef 31 c0 be 64 00 00 00 48 83 c5 01 4c 01 e3 49 81 c4 20 01 00 00 <48> 8b 0b e8 56 b3 ea ff 48 8d 7b 08 49 89 d8 4c 89 ee 48 8d 0d 75

```

### Expected behavior

works normally before 14th april

### Steps to reproduce

netdata unable to start

### Installation method

kickstart.sh

### System info

```shell
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal
```

### Netdata build info

```shell
Version: netdata v1.38.0-380-nightly
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
Debug/Developer Features:
    Trace Allocations:       NO
```

**Comment by Woart** [comment]
```
# rm /etc/netdata/netdata.conf
# service netdata start
# ps ax|grep netdata
1957757 ?        SNsl   0:01 /usr/sbin/netdata -D -P /var/run/netdata/netdata.pid
1957767 ?        SNl    0:00 /usr/sbin/netdata --special-spawn-server
1957953 ?        SNl    0:11 /usr/libexec/netdata/plugins.d/apps.plugin 1
1957955 ?        SNl    0:00 /usr/libexec/netdata/plugins.d/go.d.plugin 1
1957959 ?        SNl    0:00 /usr/libexec/netdata/plugins.d/ebpf.plugin 1
1957965 ?        SN     0:00 /usr/libexec/netdata/plugins.d/nfacct.plugin 1


```

**Comment by ilyam8** [comment]
Hi, @Woart. Thanks for reporting. We are aware of the problem (see [Past Incidents](https://status.netdata.cloud/)). The bug has been fixed and the latest nightly release contains the fix. Please, update your Netdata instance (run the updater script manually).

**Comment by Woart** [comment]
```
# wget http://localhost:19999/netdata.conf
# service netdata start

```

re-creating config solves problem

**Comment by ilyam8** [comment]
fixed in #14908
